lib/version_checker.rb

```diff
@@ -10,58 +10,50 @@ class HttpError < RuntimeError
 
 class VersionChecker
   def print_version_discrepancies
-    discrepancies_found = false
-
-    fetch_all_govuk_gemspecs.each do |gemspec|
-      rubygems_version = fetch_rubygems_version(gemspec.name)
-      unless gemspec.version == rubygems_version
-        discrepancies_found = true
-        puts "Version mismatch for '#{gemspec.name}':"
-        puts "  Version on RubyGems: #{rubygems_version}"
-        puts "  Version on GitHub: #{gemspec.version}"
-      end
-    end
+    fetch_all_govuk_gems.group_by { |gem| gem["team"] }.each do |team, gems|
+      message = gems.filter_map do |gem|
+        repo_name = gem["app_name"]
+        rubygems_version = fetch_rubygems_version(repo_name)
+        if !rubygems_version.nil? &&
+          files_changed_since_tag(repo_name, "v" + rubygems_version).any? { |path| path_is_built_into_gem?(path) }
+          "  #{repo_name} has unreleased changes since v#{rubygems_version}"
+        end
+      end.join("\n")
 
-    unless discrepancies_found
-      puts "No discrepancies found!"
+      puts team
+      puts message
     end
   end
 
-  def fetch_all_govuk_gemspecs
+  def fetch_all_govuk_gems
```

Comment:
```suggestion
  def all_govuk_gems
```
Names of ruby methods typically describe what they return.

Reply:
Great spot, adjusted.